libs/client-api/src/http.rs

```diff
@@ -52,7 +55,35 @@ use gotrue_entity::dto::SignUpResponse::{Authenticated, NotAuthenticated};
 use gotrue_entity::dto::{GotrueTokenResponse, UpdateGotrueUserParams, User};
 use realtime_entity::realtime_proto::HttpRealtimeMessage;
 
-pub const CLIENT_API_VERSION: &str = "0.0.2";
+pub const CLIENT_API_VERSION: &str = "0.0.3";
+pub const X_COMPRESSION_TYPE: &str = "X-Compression-Type";
+pub const X_COMPRESSION_BUFFER_SIZE: &str = "X-Compression-Buffer-Size";
+pub const X_COMPRESSION_TYPE_BROTLI: &str = "brotli";
+
+#[derive(Clone)]
+pub struct ClientConfiguration {
+  compression_quality: u32,
+  /// A larger buffer size means more data is compressed in a single operation, which can lead to better compression ratios
+  /// since Brotli has more data to analyze for patterns and repetitions.
+  compression_buffer_size: usize,
+}
+
+impl ClientConfiguration {
+  pub fn with_compression_buffer_size(mut self, compression_buffer_size: usize) -> Self {
+    self.compression_buffer_size = compression_buffer_size;
+    self
+  }
+}
+
+impl Default for ClientConfiguration {
+  fn default() -> Self {
+    Self {
+      compression_quality: 10,
+      compression_buffer_size: 10240,
```

Comment:
if these variables are changed after deployment and server restarts, will the data become invalid?

Reply:
Nope. Each request will carry the header values that contains the buffer size. The buffer size will be used to decompress the data.